### ISSUE TITLE
Low performance in The Incredibles Adventures of Van Helsing III DX9

### ISSUE THREAD

**Issue by mrdeathjr28** [doitsujin/dxvk#1721]
Low performance in The Incredibles Adventures of Van Helsing III DX9 with DXVK 1.7 and lastest master

Software information

The Incredibles Adventures of Van Helsing III

System information

Nvidia Drivers 450.56.06 (run package)

Lubuntu 20.04 x64 - Kernel 5.4.0-14 generic (ubuntu mainline)

CPU: Core i3 8350K Tri-Core (Coffelake 14nm) 5.0Ghz + Zalman CNPS 10x Performa+ + MasterFan Pro Air Balance 83 CFM

MEMORY: 12GB DDR4 2400mhz (8x1 - 4x1) Mushkin (dual channel - flex mode: 37.5 gb/s)

GPU: Gigabyte Nvidia Geforce GTX 1050 OC (GP107 14nm: 640 Shaders / 40 TMUS / 32 ROPS) Windforce 2GB DDR5 7000Mhz 128Bit (110Gb/s)

MAINBOARD: ASUS Z370-P

Apitrace file(s)

https://mega.nz/file/iY9RybpK#s2zY3jcWgy6l960Q8RCvHFpdnModM7AHC5np_tja8n4

Log files

[VanHelsing_x86_d3d9.log](https://github.com/doitsujin/dxvk/files/5066938/VanHelsing_x86_d3d9.log)

**Comment by doitsujin** [comment]
Can you plesase elaborate on what "low performance" means? Does wined3d work any better?

**Comment by mrdeathjr28** [comment]
> Can you plesase elaborate on what "low performance" means? Does wined3d work any better?

runs with very low performance aka in battle falls around to 20fps with postprocess, without postprocess up to 30fps with d9vk (actually dxvk)

with wined3d falls to 10fps in battle

this game must be have strange error, game have various fps lock but i use unlimited lock and others lock dont affect

**Comment by doitsujin** [comment]
and is it any better on Windows?

**Comment by mrdeathjr28** [comment]
> and is it any better on Windows? Low performance is only something we can fix if it's not just a game bug.

yeah in windows runs better

**Comment by Blisto91** [comment]
So i've tested this a bit on both a amd R9 380  with 21.3 and a nvidia 960 with 470.103.
I'm not seeing anything suspicious about the performance (without having tried the windows version) besides what might be general overhead. There is fps drops in battles yes, but not anything that makes it totally unplayable and it's seems more like the game is being weird.
When i compare  the games opengl mode with it's dx9 and dx11 on nvidia it also seems to happen there when it doesn't go through dxvk. Also when the fps drops gpu utilization isn't at 100% percent, so i think something else might be going on.
It might have improved a bit in newer dxvk versions, but not totally sure. dx11 seems to be most stable and the game might benefit from a fsync enabled kernel? Haven't tested enough
Also try to disable the setting the sync CPU and GPU something in the game (cant remember the name) as it seems to reduce performance alot.

I think more it's the game rather than dxvk being weird.

**Comment by Blisto91** [comment]
@mrdeathjr28 Friendly ping. Are you able to retest this with latest dxvk version and drivers? :slightly_smiling_face: 
Performance should have improved since.

**Comment by mrdeathjr28** [comment]
> @mrdeathjr28 Friendly ping. Are you able to retest this with latest dxvk version and drivers? slightly_smiling_face Performance should have improved since.

I dont have nvidia gpu anymore but maybe can test with my igpu uhd 630 (waiting for arc soon™)

**Comment by Blisto91** [comment]
@mrdeathjr28 Are you available to check this again? 👀

**Comment by mrdeathjr28** [comment]
> @mrdeathjr28 Are you available to check this again? eyes

yeah maybe can try but with my actual uhd 630 performance could appear low because uhd 630 lack of horsepower, however i have a try this days

**Comment by WinterSnowfall** [comment]
I gave the "Final Cut" version of the game a go (with d3d9) on an Intel Xe iGPU and can't say I've noticed much slowdowns in battles. The main performance hit seems to be the overall geometry of the landscape you're viewing.

@mrdeathjr28 Have you had a chance to confirm if this is still an issue for you?

**Comment by Blisto91** [comment]
@mrdeathjr28 Friendly ping

**Comment by mrdeathjr28** [comment]
I test (GOG version no final cut - DX9 32bits) and stay around 55fps however need test more deeply, some battles after borgovia are more heavy

![dxvk241a](https://github.com/user-attachments/assets/bff8e46b-1dd8-4ab3-8f86-e9de9cc9e8e2)

![dxvk241c](https://github.com/user-attachments/assets/671ea629-900e-47bf-a9d6-1f793e77b347)

![dxvk241d](https://github.com/user-attachments/assets/e46c3d99-f350-466c-8d3a-b6664168e187)

![dxvk241f](https://github.com/user-attachments/assets/c33300c0-24f5-43db-9bed-f69a466a292d)

At simple seek seems better

**Comment by Blisto91** [comment]
@mrdeathjr28 Did you have time to test this more deeply?

**Comment by mrdeathjr28** [comment]
> [@mrdeathjr28](https://github.com/mrdeathjr28) Did you have time to test this more deeply?

yeah begin may add a big update in channel, van helsing stay on this update, as commented before seems work ok (around 2 hours of gaming in various locations in borgovia)

https://www.youtube.com/watch?v=R5IOyr9h1H8
